### ISSUE TITLE
Create a GitHub Action to ensure that when a change is made to the main branch, that the commit hash is referenced as the version.

### ISSUE THREAD

**Issue by ehearneRedHat** [Kuadrant/kuadrantctl#82]
### What:

For non-releases, on `main` branch, when there is a new change made, a GitHub action should pick this up and change the version in `version/version.go` to something along the lines of `dev - "commit hash"` so that when the user builds the binary and runs `./kuadrantctl version` it reports:

```json
{"level":"info","ts":"2024-05-20T12:54:31+01:00","msg":"kuadrantctl version: dev - <commit hash>"}
```

**Comment by R-Lawton** [comment]
What are you seeing as non-releases? Could you give more info? 
If its what I'm thinking which is just when something is merged to main at any given time. If we automate adding the latest commit hash to version that can get very noisy and also would the latest commit not then be the commit of the gh action resulting in a new commit causing the action to be triggered again and rinse and repeat ?

**Comment by ehearneRedHat** [comment]
@R-Lawton it seems like interest is ramping up for making this happen. It might be a good idea to research it further to see if it is possible. Therefore, I am going to re-open. :)

**Comment by ehearneRedHat** [comment]
@R-Lawton I believe it is possible to do, but not as a GitHub Action. We could use `git rev-parse HEAD` to get the hash of the latest commit, which would run the command as the user runs `./kuadrantctl version` and would therefore show the user where exactly their version of the code is coming from.

**Comment by ehearneRedHat** [comment]
For example, we could implement this code in `version/version.go` in order to show the user what version they are on.

```golang
func getCommitHash() (string, error) {
    cmd := exec.Command("git", "rev-parse", "HEAD")
    output, err := cmd.Output()
    if err != nil {
        return "v0.0.0", err
    }
    commitHash := strings.TrimSpace(string(output))
    return commitHash, nil
}
version, err := getCommitHash()

if err != nil { 
  // implement code logic here to print v0.0.0 for systems without git installed
}
// implement code logic here to print commit hash for systems with git installed
```

**Comment by ehearneRedHat** [comment]
WIP https://github.com/Kuadrant/kuadrantctl/pull/85

**Comment by ehearneRedHat** [comment]
Closing as resolved in #85 .
